01-data-structures/02-stacks-and-queues/myqueue/myqueue.rb

```diff
@@ -5,14 +5,44 @@ class MyQueue
   def initialize
     @queue = Array.new
     @head = @queue[0]
+    @tail = @queue[-1]
   end
 
   def enqueue(element)
+    new_array = []
+    new_array[0] = element
+
+    @queue.each do |element|
+      new_array << element
+    end
+
+    @queue = new_array
+
+    @head = @queue[0]
+    @tail = @queue[-1]
   end
 
   def dequeue
+    # @queue.delete_at(-1)
+
+    new_array = []
+    i = 0
+    while i < @queue.length - 2
+      new_array << @queue[i]
+      i += i
+    end
+    
+    @queue = new_array
+
+    @head = @queue[0]
+    @tail = @queue[-1]
   end
 
   def empty?
+    if @queue.length == 0
```

Comment:
Oh, I see. Since this will return the last statement, I can just do this:
`def empty?
  @queue.length == 0
end`